lib/data_migrate/database_tasks.rb

```diff
@@ -115,5 +131,87 @@ def self.past_migrations(sort = nil)
 
       sort&.downcase == "asc" ? sort_migrations(migrations) : sort_migrations(migrations).reverse
     end
+
+    def self.migrate_with_data
+      DataMigrate::DataMigrator.create_data_schema_table
+
+      ActiveRecord::Migration.verbose = ENV["VERBOSE"] ? ENV["VERBOSE"] == "true" : true
+      target_version = ENV["VERSION"] ? ENV["VERSION"].to_i : nil
+      migrations = []
+
+      if target_version.nil?
+        migrations = DataMigrate::DatabaseTasks.pending_migrations.map { |m| m.merge(direction: :up) }
+      else
+        current_schema_version = ActiveRecord::Migrator.current_version
+        schema_migrations = if target_version > current_schema_version
+          DataMigrate::DatabaseTasks.pending_schema_migrations.keep_if { |m| m[:version] <= target_version }.map { |m| m.merge(direction: :up) }
+        elsif target_version < current_schema_version
+          DataMigrate::DatabaseTasks.past_migrations.keep_if { |m| m[:version] > target_version }.map { |m| m.merge(direction: :down) }
+        else # ==
+          []
+        end
+
+        current_data_version = DataMigrate::DataMigrator.current_version
+        data_migrations = if target_version > current_data_version
+          DataMigrate::DatabaseTasks.pending_data_migrations.keep_if { |m| m[:version] <= target_version }.map { |m| m.merge(direction: :up) }
+        elsif target_version < current_data_version
+          DataMigrate::DatabaseTasks.past_migrations.keep_if { |m| m[:version] > target_version }.map { |m| m.merge(direction: :down) }
+        else # ==
+          []
+        end
+        migrations = if schema_migrations.empty?
+          data_migrations
+        elsif data_migrations.empty?
+          schema_migrations
+        elsif target_version > current_data_version && target_version > current_schema_version
+          DataMigrate::DatabaseTasks.sort_migrations data_migrations, schema_migrations
+        elsif target_version < current_data_version && target_version < current_schema_version
+          DataMigrate::DatabaseTasks.sort_migrations(data_migrations, schema_migrations).reverse
+        elsif target_version > current_data_version && target_version < current_schema_version
+          schema_migrations + data_migrations
+        elsif target_version < current_data_version && target_version > current_schema_version
+          schema_migrations + data_migrations
+        end
+      end
+
+      migrations.each do |migration|
+        DataMigrate::DatabaseTasks.run_migration(migration, migration[:direction])
+      end
+    end
+
+    def self.prepare_all_with_data
+      seed = false
+
+      each_current_configuration(env) do |db_config|
```

Comment:
Hi @Floppy, thank you so much for this PR and apologies for such a late review. 

At the current version of data-migrate, we do not support data migrations on non-primary databases so iterating over each configuration here will break applications with multiple databases. However, an easy fix is to add 

```ruby
next unless db_config.primary?
```
```suggestion
      each_current_configuration(env) do |db_config|
        next unless db_config.primary?
        
```